app/javascript/omdb.js

```diff
@@ -1,38 +1,61 @@
-var OMDB_URL = 'https://private.omdbapi.com/?';
-var API_KEY = "placeholder";
+var RT_URL = "https://www.rottentomatoes.com/api/private/v2.0/search/";
+var OMDB_URL = "https://private.omdbapi.com/?";
+var OMDB_API_KEY = "placeholder";
 var TIMEOUT = 3000;
 
 var fetchedCache = {};
 var fetchingCache = {};
 
 function fetchRatings(title, season, episode, year, callback) {
-	argsString = Array.from(arguments).slice(0, 4).join(", ");
+	var argsString = Array.from(arguments).slice(0, 4).join(", ");
 	var cacheKey = hashKey(title, season, episode, year);
 	if (fetchedCache[cacheKey]) {
 		log("Cached ratings for " + argsString);
 		callback(fetchedCache[cacheKey]);
 	} else if (!fetchingCache[cacheKey]) {
-		log("Fetching ratings for " + argsString);
+		log("Fetching OMDB ratings for " + argsString);
 		fetchingCache[cacheKey] = true;
 		$.ajax({
 			url: OMDB_URL,
-			dataType: 'json',
+			dataType: "json",
 			data: requestOptions(title, season, episode, year),
 			success: function(response) {
 				if (!response.imdbRating && year) {
-					log("Failed to fetch ratings for " + argsString);
+					log("Failed to fetch OMDB ratings for " + argsString);
 					delete fetchingCache[cacheKey];
 					return fetchRatings(title, season, episode, null, callback);
 				}
 				var ratings = {
 					imdb: response.imdbRating,
 					imdbID: response.imdbID,
-					rt: fetchRTRating(response),
+					rt: filterRTRating(response),
 					metacritic: response.Metascore
-				}
-				log("Fetched ratings for " + argsString + ": " + JSON.stringify(ratings));
-				fetchedCache[cacheKey] = ratings;
-				callback(ratings);
+				};
+				log("Fetched OMDB ratings for " + argsString + ": " + JSON.stringify(ratings));
+				log("Fetching RT ratings for " + argsString);
+				$.ajax({
+					url: RT_URL,
+					type: "GET",
+					dataType: "json",
+					data: {
+						limit: 5,
+						q: response.Title
+					},
+					success: function(RTResponse) {
+						var rtInfo = fetchRTApiInfo(RTResponse, response);
+						log("Fetched RT ratings for " + argsString + ": " + JSON.stringify(rtInfo));
+						if (rtInfo.rating) ratings.rt = rtInfo.rating;
+						if (rtInfo.url) ratings.rtUrl = rtInfo.url;
+						fetchedCache[cacheKey] = ratings;
```

Comment:
This will now only cache if both calls are successful 🤔 I wonder if we could refactor this to cache OMDB and RT results separately. So first step would check for OMDB cache and grab either the result of that or make a new OMDB call and cache the result on success, then pass those ratings on to a similar function that checks RT cache or makes a new RT call, updating the RT cache on success.